lbrynet/lbrynet_daemon/Daemon.py

```
@@ -956,19 +958,22 @@ def get_est_cost(self, name, size=None):
             return self.get_est_cost_using_known_size(name, size)
         return self.get_est_cost_from_name(name)
 
+    @defer.inlineCallbacks
     def _get_lbry_file_by_uri(self, name):
-        def _get_file(stream_info):
-            sd = stream_info['sources']['lbry_sd_hash']
-
-            for l in self.lbry_file_manager.lbry_files:
-                if l.sd_hash == sd:
-                    return defer.succeed(l)
-            return defer.succeed(None)
-
-        d = self._resolve_name(name)
-        d.addCallback(_get_file)
-
-        return d
+        try:
+            stream_info = yield self._resolve_name(name)
+            sd_hash = stream_info['sources']['lbry_sd_hash']
+        except UnknownNameError:
+            sd_hash = None
+
+        if not sd_hash:
+            for lbry_file in self.lbry_file_manager.lbry_files:
+                if name == lbry_file.uri:
+                    defer.returnValue(lbry_file)
+        else:
```

changed it to raise an exception instead of returning None